### PR TITLE
scene.proto: add shadow_caster_material_name

### PR DESCRIPTION
**Pull request by scpeters**
# 🎉 New feature

Motivated by osrf/gazebo#3048.

## Summary

We added a custom SDFormat parameter in osrf/gazebo#3048 to allow specifying the name of a custom shadow caster material for a scene. Given that gazebo classic communicates scene information from the server to the client over a transport interface, it would have been convenient to add this field to the `scene.proto` message, but this would have broken ABI. As such, we added a separate ignition-transport service for this parameter.

Before ABI freezes for fortress, I thought I would submit this change, in case we decide to use the parameter in ignition in the future.

cc @iche033 @WilliamLewww 

## Test it

It's not used yet; this is just in anticipation of a future use of this parameter.

## Checklist
- [X] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

### CODE REVIEW

**Review by caguero — APPROVED** [review]

**Comment by iche033** [comment]
one suggestion is to use a script for shadow caster:

```
  message Script
  {
    repeated string uri = 1;
    string name = 2;
  }

...

  Script shadow_caster_script = 13;

```

This allows users to specify a path to the material script in case it's not in the resource paths already.

**Comment by scpeters** [comment]
> one suggestion is to use a script for shadow caster:
> 
> ```
>   message Script
>   {
>     repeated string uri = 1;
>     string name = 2;
>   }
> 
> ...
> 
>   Script shadow_caster_script = 13;
> ```
> 
> This allows users to specify a path to the material script in case it's not in the resource paths already.

@WilliamLewww does that make sense to you?

**Comment by WilliamLewww** [comment]
> > one suggestion is to use a script for shadow caster:
> > ```
> >   message Script
> >   {
> >     repeated string uri = 1;
> >     string name = 2;
> >   }
> > 
> > ...
> > 
> >   Script shadow_caster_script = 13;
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > This allows users to specify a path to the material script in case it's not in the resource paths already.
> 
> @WilliamLewww does that make sense to you?

I think in this case, the **string name** would refer to the shadow caster material name and the **string url** would refer to the file that contains the material right?

**Comment by scpeters** [comment]
> > > one suggestion is to use a script for shadow caster:
> > > ```
> > >   message Script
> > >   {
> > >     repeated string uri = 1;
> > >     string name = 2;
> > >   }
> > > 
> > > ...
> > > 
> > >   Script shadow_caster_script = 13;
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > This allows users to specify a path to the material script in case it's not in the resource paths already.
> > 
> > 
> > @WilliamLewww does that make sense to you?
> 
> I think in this case, the **string name** would refer to the shadow caster material name and the **string url** would refer to the file that contains the material (just in case the material file is not in a default path) right?

yeah, perhaps we should rename `name` to `material_name`?

**Comment by scpeters** [comment]
> > > > one suggestion is to use a script for shadow caster:
> > > > ```
> > > >   message Script
> > > >   {
> > > >     repeated string uri = 1;
> > > >     string name = 2;
> > > >   }
> > > > 
> > > > ...
> > > > 
> > > >   Script shadow_caster_script = 13;
> > > > ```
> > > > 
> > > > 
> > > >     
> > > >       
> > > >     
> > > > 
> > > >       
> > > >     
> > > > 
> > > >     
> > > >   
> > > > This allows users to specify a path to the material script in case it's not in the resource paths already.
> > > 
> > > 
> > > @WilliamLewww does that make sense to you?
> > 
> > 
> > I think in this case, the **string name** would refer to the shadow caster material name and the **string url** would refer to the file that contains the material (just in case the material file is not in a default path) right?
> 
> yeah, perhaps we should rename `name` to `material_name`?

added the `Script` sub-message in https://github.com/ignitionrobotics/ign-msgs/pull/179/commits/6addaa4080a621914c341d03d28b5ed6a585f311

File: proto/ignition/msgs/scene.proto [hunk 1]
```diff
@@ -34,6 +34,12 @@ import "ignition/msgs/header.proto";
 
 message Scene
 {
+  message Script
+  {
+    repeated string uri = 1;
+    string material_name = 2;
+  }
```
Comment (chapulina) [inline proto/ignition/msgs/scene.proto:41]: Can we document the new message and its fields?

File: proto/ignition/msgs/scene.proto [hunk 2]
```diff
@@ -51,4 +57,7 @@ message Scene
 
   /// \brief Show/hide world origin indicator.
   bool origin_visual  = 12;
+
+  /// \brief Name of shadow caster material.
```
Comment (chapulina) [inline proto/ignition/msgs/scene.proto:61]:
```suggestion
  /// \brief Shadow caster material script.
```

**Review by chapulina — COMMENTED** [review]
I don't know if it would be a good idea to use it here, but I'll just point out that we already have a `Material::Script` message here:

https://github.com/ignitionrobotics/ign-msgs/blob/2f2d86edbcbeec3e66a553608787f5792858c76e/proto/ignition/msgs/material.proto#L40-L44

File: proto/ignition/msgs/scene.proto (hunk 2 above)
Comment (scpeters) [reply proto/ignition/msgs/scene.proto]: applied in https://github.com/scpeters/ign-msgs/commit/e418f2f2d3b62e45b9bcc4f2cd5c5825f8108ab4

**Comment by scpeters** [comment]
> I don't know if it would be a good idea to use it here, but I'll just point out that we already have a `Material::Script` message here:
> 
> https://github.com/ignitionrobotics/ign-msgs/blob/2f2d86edbcbeec3e66a553608787f5792858c76e/proto/ignition/msgs/material.proto#L40-L44

I've switched to using the existing `Material.Script` message in https://github.com/scpeters/ign-msgs/commit/e418f2f2d3b62e45b9bcc4f2cd5c5825f8108ab4

**Review by chapulina — APPROVED** [review]

**Comment by chapulina** [comment]
The suggested changes have been made, I think that everyone is onboard with the current implementation. I'll merge this but speak up if you don't agree, we still have until the end of code freeze to revert it if needed.

**Pull request merged**